src/mesh/voxel_cloud.rs

```diff
@@ -1267,6 +1384,74 @@ impl ScalarField {
         }
     }
 
+    /// Add values of the other scalar fields to the respective voxels of the
+    /// current scalar field. The current scalar field will not be resized, so
+    /// only the overlapping areas will be will be added. If the voxel in the
+    /// current or the other scalar field is None (does not contain any voxel or
+    /// does not exist), it is treated according to the selected
+    /// `empty_voxels_treatment` for both, the current and the other scalar
+    /// field - it is either replaced with a constant value or considered None
+    /// and then also the resulting voxel will become None.
+    // FIXME: Join with interpolate_to
+    pub fn add_values(
+        &mut self,
+        other: &ScalarField,
+        empty_voxels_treatment_current: TreatEmpty,
+        empty_voxels_treatment_other: TreatEmpty,
+    ) {
+        for (one_dimensional, voxel) in self.voxels.iter_mut().enumerate() {
+            let cartesian_coordinate = one_dimensional_to_cartesian_coordinate(
+                one_dimensional,
+                &self.block_start,
+                &self.block_dimensions,
+                &self.voxel_dimensions,
+            );
+            let absolute_coordinate_other = cartesian_to_absolute_voxel_coordinate(
+                &cartesian_coordinate,
+                &other.voxel_dimensions,
+            );
+            let voxel_other = other.value_at_absolute_voxel_coordinate(&absolute_coordinate_other);
+
+            *voxel = match empty_voxels_treatment_current {
+                TreatEmpty::AsNone => {
+                    if let Some(value_self) = &voxel {
+                        match empty_voxels_treatment_other {
+                            TreatEmpty::AsNone => {
+                                if let Some(value_other) = voxel_other {
+                                    Some(value_self + value_other)
+                                } else {
+                                    None
```

Comment:
Is this how we want it to behave? `Some(value_self)` also looks like a legit alternative.

Reply:
Not really, at least not when then treatment is set to `AsNone`. I can imagine adding something like `Ignore` here, that would do what you request

Reply:
I run into the same issue with Grasshopper today. With each new version of Grasshopper (which doesn't happen too often), there are some new versions of components. However, if you open a file with old components, Grasshopper uses the old versions of the components so that you get the same result as before. 

Now the old `Multiply component` returned `Null` for `Null * Number` and a `Vector` for `Vector * Number`. The new `Multiply component` returns a `Number` for  `Null * Number` (`Null * 1.0 = 1.0`). Both, old and new Grasshopper return a `Vector` for `Null + Vector` but fail for `Number + Vector`. So a chain `Null * Number + Vector` used to return `Vector` and now it fails. 

The moral of this story is: there is no convention ;)